Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
-The fee rate to be used when computing fees for the different transactions.
+#### Requirements
 
-#### CET CSV Delay
-The number to use as input to the CSV in the [CETs](Transactions.md#outputs-1).
+The sending node MUST:
 
-#### CET lock time
-The number to use as `nLockTime` for the CET transactions.
-This should correspond to the maturity date of the contracts, invalidating early CET broadcasting.
+  - set undefined bits in `contract_flags` to 0.
+  - ensure the `chain_hash` value identifies the chain it wishes to open the contract within.
+  - set `funding_pubkey` to a valid secp256k1 pubkey in compressed format.
+  - set `total_collateral_satoshis` greater than or equal to 1000.
+  - set `contract_maturity_cltv` and `contract_timeout_cltv` to either both be UNIX timestamps, or both be block heights as distinguished in [BIP 65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki).
+  - set `contract_maturity_cltv` to be less than `contract_timeout_cltv`.
 
-#### Refund locktime
-The locktime to be used for the refund transaction.
-It should be set at a later date than the maturity date of the contract.
+The sending node SHOULD:
 
----
+  - set `feerate_per_vb` to at least the rate it estimates would cause the transaction to be immediately included in a block.
+  - set `contract_maturity_cltv` to the earliest expected oracle signature time.
+  - set `contract_timeout_cltv` sufficiently long after `contract_maturity_cltv` to allow for late oracle signatures and other delays to closing the contract.
 
-### Accept
-After receiving the `offer` message from A, and after [validating](#offer-validation) the provided information, B creates and sends back an `accept` message containing:
-1. B's fund amount (satoshi)
-1. B's public keys
-1. B's addresses
-1. B's inputs
-1. CET signatures
-1. Refund signature
+The receiving node MUST:
 
-#### B's fund amount (satoshi)
-How much B inputs into the contract.
+  - ignore undefined bits in `channel_flags`.
 
-#### B's public keys
-The following public keys:
+The receiving node MAY reject the contract if:
 
-1. Funding Public Key
-   - B's public key used in the multi-signature [funding output](Transactions.md#FundingOutputs)
-2. Sweep Public Key
-   - B's public key used in the creation of the CETs [output script](Transactions.md#CETOutputs).
+  - it does not agree to the terms in `contract_info`.
+  - the `contract_info` is missing relevant events.
+  - it does not want to use the oracle(s) specified in `oracle_info`.
+  - `total_collateral_satoshis` is too small.
 
-#### B's addresses
+The receiving node MUST reject the contract if:
 
-The following addresses:
+  - the `chain_hash` value is set to a hash of a chain that is unknown to the receiver.
+  - the `contract_info` refers to events unknown to the receiver.
+  - the `oracle_info` refers to an oracle unknown to the receiver.
+  - it considers `feerate_per_vb` too small for timely processing or unreasonably large.
```

Comment:
The only time this would lead to a rejection is if the resulting funding transaction would not be valid so I'm not sure I need to say this? Will double check against Lisa's spec.

Reply:
I should have been more specific about what I meant by "valid". In this case I meant valid in terms of relay policy on the bitcoin p2p network. If you want more background on this you look at the [`AcceptToMemoryPool`](https://github.com/bitcoin/bitcoin/blob/b75f2ad72db6db93665c66279a4c9e8d5d89f027/src/validation.cpp#L1062) (ATMP) logic in bitcoin core.

and you should definitely checkout this excellent discussion from @ariard about relay policies on the bitcoin network in the future

https://github.com/bitcoin/bitcoin/issues/14895#issuecomment-665874764